src/rascal/representations/calculator_spherical_invariants.hh

```diff
@@ -254,6 +254,41 @@ namespace rascal {
     //! the coefficient_subselection input
     bool is_sparsified{false};
 
+    int get_num_coefficients(int n_species) {
+      auto soap_type = hypers.at("soap_type").get<std::string>();
+      if (soap_type == "RadialSpectrum") {
+        return (n_species * this->max_radial);
+      } else if (soap_type == "PowerSpectrum") {
+        return (n_species * (n_species + 1) * 0.5 * this->max_radial *
+                this->max_radial * (this->max_angular + 1));
```

Comment:
Could you tell me the advantage in this case. If we do not return an `int` here, we get a compile error because the output is declared as `int`

Reply:
I put it now around the species, because this should evaluate to an int. Besides removing some unnecessary floating point computation, I however do not see any advantage.

Reply:
doing the static cast here is just to specify clearly what happens implicitly with the return type.